### PR TITLE
Add selectBox Vue and React examples

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: concepts/05 Widgets/SelectBox/04 Handle the Value Change Event.md [hunk 1]
```diff
@@ -90,6 +155,64 @@ To process a new **SelectBox** value, you need to handle the value change event.
...
+
+    import { DxSelectBox } from 'devextreme-vue/select-box';
+
+    export default {
+        components: {
+            DxSelectBox
+        },
+        methods: {
+            valueChanged(e) {
+                let previousValue = e.previousValue;
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/04 Handle the Value Change Event.md]:
```suggestion
                const previousValue = e.previousValue;
```

File: concepts/05 Widgets/SelectBox/04 Handle the Value Change Event.md [hunk 2]
```diff
@@ -90,6 +155,64 @@ To process a new **SelectBox** value, you need to handle the value change event.
...
+                let newValue = e.value;
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/04 Handle the Value Change Event.md]:
```suggestion
                const newValue = e.value;
```

File: concepts/05 Widgets/SelectBox/04 Handle the Value Change Event.md [hunk 3]
```diff
@@ -90,6 +155,64 @@ To process a new **SelectBox** value, you need to handle the value change event.
...
+
+    import SelectBox from 'devextreme-react/select-box';
+
+    class App extends React.Component {
+        constructor(props) {
+            super(props);
+            this.valueChanged = this.valueChanged.bind(this);
+        }
+        valueChanged(e) {
+            let previousValue = e.previousValue;
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/04 Handle the Value Change Event.md]:
```suggestion
            const previousValue = e.previousValue;
```

File: concepts/05 Widgets/SelectBox/04 Handle the Value Change Event.md [hunk 4]
```diff
@@ -90,6 +155,64 @@ To process a new **SelectBox** value, you need to handle the value change event.
...
+            let newValue = e.value;
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/04 Handle the Value Change Event.md]:
```suggestion
            const newValue = e.value;
```

File: concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md [hunk 5]
```diff
@@ -88,6 +150,121 @@ If you need a more flexible solution, define a custom template for widget items.
...
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSelectBox ...
+            :data-source="selectBoxData"
+            display-expr="name"
+            value-expr="id"
+            item-template="item"
+        >
+            <template #item="item">
+                <Item :item-data="item.data"/>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md]: Let's keep the example simpler and get rid of the custom component.

File: concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md [hunk 6]
```diff
@@ -88,6 +150,121 @@ If you need a more flexible solution, define a custom template for widget items.
...
+    ];
+
+    class App extends React.Component {
+        render() {
+            return (
+                <SelectBox ...
+                    dataSource={selectBoxData} 
+                    displayExpr="name"
+                    valueExpr="id"
+                    itemRender={Item}
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md]: Here and in every other template examples too. Let's use a render function.

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 7]
```diff
@@ -51,6 +51,72 @@ The **SelectBox** widget allows an end user to search through its items. This fe
...
+            DxSelectBox
+        },
+        data() {
+            const selectBoxData = [
+                { id: 1, country: "Afghanistan", capital: "Kabul" },
+                { id: 2, country: "Albania", capital: "Tirana" },
+                // ...
+            ];
+            return {
+                selectBoxData,
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
                selectBoxData
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 8]
```diff
@@ -51,6 +51,72 @@ The **SelectBox** widget allows an end user to search through its items. This fe
          // ...
      })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSelectBox ...
+            :data-source="selectBoxData"
+            :searchEnabled="true"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
            :search-enabled="true"
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 9]
```diff
@@ -51,6 +51,72 @@ The **SelectBox** widget allows an end user to search through its items. This fe
...
+            :searchExpr="['country', 'capital']"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
            :search-expr="['country', 'capital']"
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 10]
```diff
@@ -104,6 +176,74 @@ When a user types a string in the input field, the **SelectBox** suggests all it
          // ...
      })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSelectBox ...
+            :data-source="selectBoxData"
+            :searchEnabled="true"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
            :search-enabled="true"
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 11]
```diff
@@ -104,6 +176,74 @@ When a user types a string in the input field, the **SelectBox** suggests all it
...
+            searchExpr="country"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
            search-expr="country"
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 12]
```diff
@@ -104,6 +176,74 @@ When a user types a string in the input field, the **SelectBox** suggests all it
...
+            DxSelectBox
+        },
+        data() {
+            const selectBoxData = [
+                { id: 1, country: "Afghanistan", capital: "Kabul" },
+                { id: 2, country: "Albania", capital: "Tirana" },
+                // ...
+            ];
+            return {
+                selectBoxData,
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
                selectBoxData
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 13]
```diff
@@ -153,6 +299,74 @@ There is a delay between the moment a user finishes typing and the moment the se
          // ...
      })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSelectBox ...
+            :data-source="selectBoxData"
+            :searchEnabled="true"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
            :search-enabled="true"
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 14]
```diff
@@ -153,6 +299,74 @@ There is a delay between the moment a user finishes typing and the moment the se
...
+            :search-timeout="1000"
+            searchExpr="country"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
            search-expr="country"
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 15]
```diff
@@ -153,6 +299,74 @@ There is a delay between the moment a user finishes typing and the moment the se
...
+            DxSelectBox
+        },
+        data() {
+            const selectBoxData = [
+                { id: 1, country: "Afghanistan", capital: "Kabul" },
+                { id: 2, country: "Albania", capital: "Tirana" },
+                // ...
+            ];
+            return {
+                selectBoxData,
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
                selectBoxData
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 16]
```diff
@@ -201,6 +420,72 @@ The **SelectBox** widget starts searching after a user has typed at least one ch
          // ...
      })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSelectBox ...
+            :data-source="selectBoxData"
+            :searchEnabled="true"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
            :search-enabled="true"
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 17]
```diff
@@ -201,6 +420,72 @@ The **SelectBox** widget starts searching after a user has typed at least one ch
...
+            :min-search-length="3"
+            searchExpr="country"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
            search-expr="country"
```

File: concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md [hunk 18]
```diff
@@ -201,6 +420,72 @@ The **SelectBox** widget starts searching after a user has typed at least one ch
...
+        components: {
+            DxSelectBox
+        },
+        data() {
+            const selectBoxData = [
+                { id: 1, country: "Afghanistan", capital: "Kabul" },
+                // ...
+            ];
+            return {
+                selectBoxData,
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/10 Configure Search Parameters.md]:
```suggestion
                selectBoxData
```

File: concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md [hunk 19]
```diff
@@ -75,6 +75,101 @@ A user can select existing values and add new values to the **SelectBox**. To en
...
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSelectBox ...
+            :data-source="selectBoxData"
+            :accept-custom-value="true"
+            display-expr="firstName"
+            value-expr="id"
+            @customItemCreating="customItemCreating($event)"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md]:
```suggestion
            @custom-item-creating="customItemCreating"
```

File: concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md [hunk 20]
```diff
@@ -75,6 +75,101 @@ A user can select existing values and add new values to the **SelectBox**. To en
...
+
+    import { DxSelectBox } from 'devextreme-vue/select-box';
+    import DataSource from "devextreme/data/data_source";
+
+    const selectBoxData = new DataSource({
+        store: [
+            { id: 1, firstName: "Andrew" },
+            { id: 2, firstName: "Nancy" },
+            { id: 3, firstName: "Steven" }
+        ]
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md]:
```suggestion
        ],
        key: "id"
```

File: concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md [hunk 21]
```diff
@@ -75,6 +75,101 @@ A user can select existing values and add new values to the **SelectBox**. To en
...
+        },
+        data() {
+            return {
+                selectBoxData
+            }
+        }
+        methods: {
+            customItemCreating(e) {
+                // Generates a new 'id'
+                var nextId = Math.max.apply(Math, selectBoxData.items().map(function(c) { return c.id; })) + 1;
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md]:
```suggestion
                const nextId = Math.max.apply(Math, selectBoxData.items().map(c => c.id)) + 1;
```

File: concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md [hunk 22]
```diff
@@ -75,6 +75,101 @@ A user can select existing values and add new values to the **SelectBox**. To en
...
+            { id: 1, firstName: "Andrew" },
+            { id: 2, firstName: "Nancy" },
+            { id: 3, firstName: "Steven" }
+        ]
+    });
+
+    class App extends React.Component {
+        customItemCreating(e) {
+            // Generates a new 'id'
+            var nextId = Math.max.apply(Math, selectBoxData.items().map(function(c) { return c.id; })) + 1;
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md]:
```suggestion
            const nextId = Math.max.apply(Math, selectBoxData.items().map(c => c.id)) + 1;
```

File: concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md [hunk 23]
```diff
@@ -75,6 +75,101 @@ A user can select existing values and add new values to the **SelectBox**. To en
...
+
+    import SelectBox from 'devextreme-react/select-box';
+    import DataSource from "devextreme/data/data_source";
+
+    const selectBoxData = new DataSource({
+        store: [
+            { id: 1, firstName: "Andrew" },
+            { id: 2, firstName: "Nancy" },
+            { id: 3, firstName: "Steven" }
+        ]
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/15 Create a User-Defined Item.md]:
```suggestion
        ],
        key: "id"
```

File: concepts/05 Widgets/SelectBox/25 Keyboard Support.md [hunk 24]
```diff
@@ -81,6 +81,72 @@ You can implement a custom handler for a key using the [registerKeyHandler(key,
...
+    import React from 'react';
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import SelectBox from 'devextreme-react/select-box';
+
+    class App extends React.Component {
+        constructor(props) {
+            super(props);
+            this.registerKeyHandlers = this.registerKeyHandlers.bind(this);
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/25 Keyboard Support.md]: You don't address to the state inside the `registerKeyHandlers` function, so there's no need to bind it to the component's context. You can remove the entire constructor from this example.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md [hunk 25]
```diff
@@ -258,6 +414,107 @@ Using similar techniques, you can customize the input field of the **SelectBox**
...
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSelectBox ...
+            :data-source="selectBoxData"
+            :value="1"
+            display-expr="name"
+            value-expr="id"
+            field-template="field"
+        >
+
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md]:
```suggestion

```

File: concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md [hunk 26]
```diff
@@ -258,6 +414,107 @@ Using similar techniques, you can customize the input field of the **SelectBox**
...
+            <template #field='{ data }'>
+                <div>
+                    <img
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md]:
```suggestion
                    <img :src="data.imgSrc">
```

File: concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md [hunk 27]
```diff
@@ -258,6 +414,107 @@ Using similar techniques, you can customize the input field of the **SelectBox**
...
+                        :src="data.imgSrc"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md]:
```suggestion

```

File: concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md [hunk 28]
```diff
@@ -258,6 +414,107 @@ Using similar techniques, you can customize the input field of the **SelectBox**
...
+                    >
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md]:
```suggestion

```

File: concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md [hunk 29]
```diff
@@ -258,6 +414,107 @@ Using similar techniques, you can customize the input field of the **SelectBox**
...
+                </div>
+            </template>
+        </DxSelectBox>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DxSelectBox, DxTextBox } from 'devextreme-vue';
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md]:
```suggestion
    import { DxSelectBox } from 'devextreme-vue/select-box';
    import { DxTextBox } from 'devextreme-vue/text-box';
```

File: concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md [hunk 30]
```diff
@@ -258,6 +414,107 @@ Using similar techniques, you can customize the input field of the **SelectBox**
...
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { SelectBox, TextBox } from 'devextreme-react';
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md]:
```suggestion
    import { SelectBox } from 'devextreme-react/select-box';
    import { TextBox } from 'devextreme-react/text-box';
```

File: concepts/05 Widgets/SelectBox/07 Grouping/05 Customize Group Headers.md [hunk 31]
```diff
@@ -54,6 +54,128 @@ By default, group headers display text of the **key** field in a bold font. You
...
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSelectBox ...
+            :data-source="selectBoxDataSource"
+            :grouped="true"
+            display-expr="name"
+            value-expr="count"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/07 Grouping/05 Customize Group Headers.md:65]:
```suggestion
            value-expr="count"
            group-template="group"
```

File: concepts/05 Widgets/SelectBox/07 Grouping/05 Customize Group Headers.md [hunk 32]
```diff
@@ -54,6 +54,128 @@ By default, group headers display text of the **key** field in a bold font. You
...
+            groupedItem.items.forEach(function(item) {
+                overallCount += item.count;
+            })
+            return Object.assign(groupedItem, { overallCount: overallCount });
+        }
+    });
+
+    const renderGroup = (data) => {
+        return (
+            <p >
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/07 Grouping/05 Customize Group Headers.md]:
```suggestion
            <p>
```

File: concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md [hunk 33]
```diff
@@ -88,6 +150,100 @@ If you need a more flexible solution, define a custom template for widget items.
...
+            :data-source="selectBoxData"
+            display-expr="name"
+            value-expr="id"
+            item-template="item"
+        >
+            <template #item="{ data }">
+                <div>
+                    <img :src="data.imgSrc">
+                    <div style="display:inline-block">
+                    {{ data.name }}
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/SelectBox/05 Customize Item Appearance.md]:
```suggestion
                        {{ data.name }}
```

**Review by RomanTsukanov — APPROVED** [review]

**Review by AlexJameson — APPROVED** [review]

**Pull request merged**